### PR TITLE
IfcRelAdheresToElement authoring

### PR DESCRIPTION
**Pull request by AlexBrad1eyCT**
Authoring of IfcRelAdheresToElement relationship in response to issue #28, includes the following:

1. New Entity IfcRelAdheresToElement
2.  Inverse Attribute HasSurfaceFeatures on IfcElement
3. InverseAttribute AdheresToElement on IfcSurfaceFeature

Reviews and Discussion Welcome!

### CODE REVIEW

**Comment by SergejMuhic** [comment]
I have added @TLiebich to review this one. Would like to hear his opinion.

**Comment by SergejMuhic** [comment]
Adding a suggestion based on the conversation after @larswik produced a test IFC file:

"For "my" unit test, 1:1 is ok, but in the generic case maybe one IfcElement vs many IfcSurfaceFeature could be possible."

**Comment by larswik** [comment]
An example file attached.

[RumbleStrip-INDOT.zip](https://github.com/bSI-InfraRoom/IFC-Specification/files/6662137/RumbleStrip-INDOT.zip)

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcElement/DocEntity.xml [hunk 1]
```diff
@@ -62,6 +62,9 @@
 		<DocAttribute Name="HasCoverings" UniqueId="e73f220c-19f9-4fbd-9302-9b407708ffdf" DefinedType="IfcRelCoversBldgElements" AggregationType="3" Inverse="RelatingBuildingElement" XsdFormat="element">
 			<Documentation>Reference to _IfcCovering_ by virtue of the objectified relationship _IfcRelCoversBldgElement_. It defines the concept of an element having coverings associated.</Documentation>
 		</DocAttribute>
+		<DocAttribute Name="HasSurfaceFeatures" UniqueId="f809b0e2-6c5a-447b-92e9-841a1725ea2e" DefinedType="IfcRelAdheresToElement" AggregationType="3" Inverse="RelatingElement" XsdFormat="element">
+			<Documentation>Reference to the _IfcRelAdheresToElement_ relationship that adheres a _IfcSurfaceFeature_ to an element. An element can incorporate zero-to-many surface features. For each surfatce feature, a new relationship is generated.</Documentation>
```
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcElement/DocEntity.xml]: Sorted

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/DocSchema.xml [hunk 2]
```diff
@@ -164,6 +164,7 @@
 			<Definitions>
 				<DocDefinitionRef Name="IfcFooting" UniqueId="d2ddad6c-5d29-4a93-b302-8d8162712cb0" />
 				<DocDefinitionRef Name="IfcPile" UniqueId="ecf767f8-2182-4fe4-af66-72c57aa2b099" />
+				<DocDefinitionRef Name="IfcSurfaceFeature" UniqueId="c0e4e121-bf02-47a3-b920-ece56bee40d2" />
```
[the remark this answers is not in the document]
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/DocSchema.xml:167]: Shall we leave this as a future enhance,ent via an issue

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcRelAdheresToElement/Documentation.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+The _IfcRelAdheresToElement_ is an objectified relationship between an element and one surface feature element that that adheres to the surface on the element. The relationship is defined to be a 1 to many relationship. The _IfcRelAdheresToElement_ establishes an aggregation relationship between the main element and a sub ordinary surface feature.
```
Comment (larswik) [inline IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcRelAdheresToElement/Documentation.md:1]: Suggest "The _IfcRelAdheresToElement_ is an objectified relationship between an element and **one to many surface feature elements** that **(remove one that)**"

**Review by larswik — CHANGES_REQUESTED** [review]

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcRelAdheresToElement/Documentation.md (hunk 3 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcRelAdheresToElement/Documentation.md:1]:
```suggestion
The _IfcRelAdheresToElement_ is an objectified relationship between an element and one to many surface feature elements that adhere to the surface of the element. The relationship is defined to be a 1 to many relationship. The _IfcRelAdheresToElement_ establishes an aggregation relationship between the main element and a sub ordinary surface feature.
```

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/DocSchema.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (SergejMuhic) [reply IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/DocSchema.xml:167]: Done #121

**Review by SergejMuhic — APPROVED** [review]
I think this looks good.

**Pull request merged**